tests/NSubstitute.Acceptance.Specs/ReturningResults.cs

```diff
@@ -381,6 +381,19 @@ public void Returns_null_for_any_args_when_TaskValue_is_of_null_value_type()
         Assert.That(_something.NullableCountValueTaskWithParamsAsync(123, "something else").Result, Is.Null);
     }
 
+    [Test]
+    public void DerivedInterfaces()
+    {
+        var derived = Substitute.For<Infrastructure.DerivedInterfaces.B>();
+        var parent = derived as Infrastructure.DerivedInterfaces.A;
+
+        derived.X.Returns("configuredvalue");
+        Assert.That(parent.X, Is.EqualTo("configuredvalue"));
+
+        parent.X.Returns("newvalue");
+        Assert.That(derived.X, Is.EqualTo("newvalue"));
+    }
+
```

Comment:
**question:** `parent.X` and `derived.X` are not actually connected in this case are they? They happen to share a name, but they represent different things. (hence the `new` in `interface B` declaration.)

Based on that, I would expected `derived.X` and `parent.X` to return potentially different values, and to be stubbed independently.

Please correct me if i'm missing something here 🙇

Reply:
In the supplied example the properties are on interfaces, and when a concrete class implements `B`, there is only ever a single implementation of the `X` property. You are correct in that in non-interface scenarios there could actually be more than one implementation of the `X` property. 

The difficulty I experienced (and was attempting to solve with this PR) was that I could only configure a single `.Returns` (that was declared on the type provided to `.For<T>`). In writing this response I realise that I could have cast and configured the return again. Whilst this does muddy my setup a little, it requires no changes to NSubstitute. I am content abandon this PR.

Reply:
> In the supplied example the properties are on interfaces, and when a concrete class implements B, there is only ever a single implementation of the X property.

I think using explicit interface implementation there could be two?

> Whilst this does muddy my setup a little, it requires no changes to NSubstitute. I am content to abandon this PR.

Yeah it does make the setup a bit messier, but I think it's more correct that way. You might be able to use `ReturnsForAll` to help a bit if `X` is the only property with that type.

Thank you for the contribution and the discussion around it. Even thought it didn't get merged I think it was very useful to work through and confirm the behaviour for this case. 🙇